test/metabase/api/search_test.clj

```diff
@@ -610,3 +609,17 @@
       (is (= ["Another SQL query"]
              (->> (search-request-data :rasta :q "aggregation")
                   (map :name)))))))
+
+(deftest app-test
+  (testing "App collections should come with app_id set"
+    (with-search-items-in-collection {:keys [collection]} "test"
+      (mt/with-temp App [_app {:collection_id (:id collection)}]
+        (is (= (mapv
+                (fn [result]
+                  (cond-> result
+                    (not (#{"metric" "segment"} (:model result)))
+                    (assoc-in [:collection :app_id] true)
+                    (= (:model result) "collection")
```

Comment:
nit: I find a blank line between cond pairs are more readable when the pairs aren't on the same line.